src/data/program_configuration.js

```diff
@@ -93,36 +96,40 @@ class ConstantBinder<T> implements Binder<T> {
                 currentValue: PossiblyEvaluatedPropertyValue<T>) {
         const value: any = currentValue.constantOr(this.value);
         const gl = context.gl;
-        if (this.type === 'color') {
-            gl.uniform4f(program.uniforms[`u_${this.name}`], value.r, value.g, value.b, value.a);
-        } else {
-            gl.uniform1f(program.uniforms[`u_${this.name}`], value);
-        }
+        this.names.forEach((name)=>{
+            if (this.type === 'color') {
+                gl.uniform4f(program.uniforms[`u_${name}`], value.r, value.g, value.b, value.a);
+            } else {
+                gl.uniform1f(program.uniforms[`u_${name}`], value);
+            }
+        });
     }
 }
 
 class SourceExpressionBinder<T> implements Binder<T> {
     expression: SourceExpression;
-    name: string;
+    names: Array<string>;
     type: string;
     statistics: { max: number };
 
     paintVertexArray: StructArray;
     paintVertexAttributes: Array<StructArrayMember>;
     paintVertexBuffer: ?VertexBuffer;
 
-    constructor(expression: SourceExpression, name: string, type: string) {
+    constructor(expression: SourceExpression, names: Array<string>, type: string, layout: () => StructArray) {
         this.expression = expression;
-        this.name = name;
+        this.names = names;
         this.type = type;
         this.statistics = { max: -Infinity };
-        const PaintVertexArray = type === 'color' ? StructArrayLayout2f8 : StructArrayLayout1f4;
-        this.paintVertexAttributes = [{
-            name: `a_${name}`,
-            type: 'Float32',
-            components: type === 'color' ? 2 : 1,
-            offset: 0
-        }];
+        const PaintVertexArray = layout;
+        this.paintVertexAttributes = names.map((name)=> {
+            return {
+                name: `a_${name}`,
+                type: 'Float32',
+                components: type === 'color' ? 2 : 1,
+                offset: 0
+            };
+        });
```

Comment:
Nit: you can omit braces and `return` here too if you surround the object with `(...)`.